context_engineering_functions.py

```diff
@@ -0,0 +1,91 @@
+import pandas as pd
+import os
+
+def get_available_maps(df):
+    df.sort_values(by=['MatchId', 'DecisionOrder'], ascending=True, inplace=True)
+    df.reset_index(drop=True, inplace=True)
+    
+    map_names = df['MapName'].unique().tolist()
+
+    # get which map we're talking about, OHE
+    manip_df = pd.concat([df[['MatchId']], pd.get_dummies(df['MapName'])], axis=1)
+
+    # 1 - what's not available
+    rolling_df = (1 - (manip_df.groupby('MatchId')[map_names]
+                               .rolling(7, min_periods=0)
+                               .sum()  # rolling sum
+                               .shift(1, fill_value=0)  # shift down so we know what's available
+                               ))
+    rolling_df = rolling_df.astype(int)
+    rolling_df.reset_index(drop=True, inplace=True)
+    rolling_df.columns = [x + '_is_available' for x in rolling_df.columns]
+
+    df = pd.concat([df, rolling_df], axis=1)
+    return df
+
+def get_rewards(map_picks, demos):
+    map_picks =  pd.merge(map_picks,
+                          demos[['MatchId', 'MapName', 'WinnerId']], 
+                          how = 'left', 
+                          left_on= ['MatchId', 'MapName'], 
+                          right_on = ['MatchId', 'MapName'],
+                          suffixes = (None, '_right'))
+    #for picks, if winner is same as picker, reward 1
+    map_picks['Y_reward'] = (map_picks.DecisionTeamId == map_picks.WinnerId).astype(int)
+
+    #drop extra WinnerId column since we no longer need it
+    map_picks.dropna(inplace= True, axis = 0)
+    map_picks.drop(labels = [ 'Created', 'Updated', 'WinnerId'], axis = 1, inplace = True)
+
+    return map_picks
+
+
+def create_basic_triples(data_directory, save = False):
```

Comment:
Any idea why I'm getting an uneven distribution of DecisionOrders:

Run: `map_pick_context['DecisionOrder'].value_counts()`
Output:
```
4    4792
3    4786
7    3400
5      82
6      47
1       2
2       1
Name: DecisionOrder, dtype: int64
```

Reply:
So looking into the data, there are a few matches where they pick more than 3 maps. Should we remove them? Something to think about.